### PR TITLE
store/tikv: remove dependency on `tidb` from `store/tikv`.

### PR DESCRIPTION
**Pull request by coocood**
Makes `tidb` package able to import `store/tikv`.

### CODE REVIEW

**Comment by coocood** [comment]
@hanfei1991 @disksing @tiancaiamao PTAL

**Comment by hanfei1991** [comment]
LGTM

File: store/tikv/test_util.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+import (
+	"flag"
+	"fmt"
+	"github.com/juju/errors"
+	"github.com/pingcap/kvproto/pkg/errorpb"
+	"github.com/pingcap/tidb/store/tikv/oracle"
+	"github.com/pingcap/tidb/store/tikv/tikvrpc"
+	goctx "golang.org/x/net/context"
+	"sync"
+	"time"
```
Comment (disksing) [inline store/tikv/test_util.go]: reorder.

File: store/tikv/sql_test.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
+package tikv_test
```
Comment (disksing) [inline store/tikv/sql_test.go]: license.

File: store/tikv/gcworker/gc_worker.go [hunk 3]
```diff
@@ -512,15 +434,15 @@ func (w *GCWorker) deleteRanges(ctx goctx.Context, safePoint uint64) error {
 	return nil
 }
 
-func resolveLocks(ctx goctx.Context, store *tikvStore, safePoint uint64, identifier string) error {
+func resolveLocks(ctx goctx.Context, store tikv.Storage, safePoint uint64, identifier string) error {
```
Comment (disksing) [inline store/tikv/gcworker/gc_worker.go]: Seems we can make it a method of `tikvStore` and put it inside tikv package.

Comment (coocood) [reply store/tikv/gcworker/gc_worker.go]: We can do it in future PR.

File: store/tikv/backoff.go [hunk 4]
```diff
@@ -142,9 +143,9 @@ const (
 	getMaxBackoff           = 20000
 	prewriteMaxBackoff      = 20000
 	cleanupMaxBackoff       = 20000
-	gcMaxBackoff            = 100000
-	gcResolveLockMaxBackoff = 100000
-	gcDeleteRangeMaxBackoff = 100000
+	GcMaxBackoff            = 100000
```
Comment (zimulala) [inline store/tikv/backoff.go]: s/GcMaxBackoff/GCMaxBackoff

**Comment by disksing** [comment]
LGTM.

**Pull request merged**